### PR TITLE
Add package ownership to storage model

### PR DESCRIPTION
**Pull request by callistonianembrace**
This PR adds ownership to packages. 

It adds a field, `admin` to packages and later checks if publisher is indeed the owner of the package.

### CODE REVIEW

File: packages/api/src/publish.ts [hunk 1]
```diff
@@ -303,25 +303,34 @@ export function publishPackage(storage: Storage, config: Config, auth: IAuth): a
       const metadata = validateMetadata(req.body, packageName);
       debug('post validation metadata to publish %o', metadata);
       // treating deprecation as updating a package
+      const remote = req.remote_user;
       if (req.params._rev || isRelatedToDeprecation(req.body)) {
+        debug('=========================');
```
Comment (callistonianembrace) [inline packages/api/src/publish.ts:308]:
```suggestion
```

File: packages/api/src/publish.ts [hunk 2]
```diff
@@ -303,25 +303,33 @@ export function publishPackage(storage: Storage, config: Config, auth: IAuth): a
...
+          
           debug('update a package');
           storage.changePackage(packageName, metadata, req.params.revision, function (error) {
             afterChange(error, API_MESSAGE.PKG_CHANGED, metadata);
           });
         });
       } else {
         debug('adding a new version for the package %o', packageName);
-        storage.addPackage(packageName, metadata, function (error) {
+        debug('---------------adding %o as admin for %o', remote, packageName);
```
Comment (callistonianembrace) [inline packages/api/src/publish.ts:330]:
```suggestion
```

File: packages/api/src/publish.ts [hunk 3]
```diff
@@ -303,25 +303,32 @@ export function publishPackage(storage: Storage, config: Config, auth: IAuth): a
...
           debug('update a package');
           storage.changePackage(packageName, metadata, req.params.revision, function (error) {
             afterChange(error, API_MESSAGE.PKG_CHANGED, metadata);
           });
         });
       } else {
         debug('adding a new version for the package %o', packageName);
-        storage.addPackage(packageName, metadata, function (error) {
+        storage.addPackage(packageName, remote.name, metadata, function (error) { 
+          debug("error ------------>", error);
```
Comment (callistonianembrace) [inline packages/api/src/publish.ts:331]:
```suggestion
```

**Comment by melwyn95** [comment]
Nice, I thought verdaccio would provide this out of the box

**Pull request merged**